### PR TITLE
Apply `black` to `astropy.samp`

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Apply `black` to `astropy.samp` according to the process laid out by [APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst).

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/samp/__init__.py [hunk 1]
```diff
@@ -27,12 +27,13 @@ class Conf(_config.ConfigNamespace):
 
     use_internet = _config.ConfigItem(
         True,
-        "Whether to allow `astropy.samp` to use "
-        "the internet, if available.",
-        aliases=['astropy.samp.utils.use_internet'])
+        "Whether to allow `astropy.samp` to use " "the internet, if available.",
```
Comment (astrofrog) [inline astropy/samp/__init__.py:30]: We should fix this string concatenation (either here or in follow-up across package)

File: astropy/samp/hub.py [hunk 2]
```diff
@@ -943,14 +1046,22 @@ def _notify(self, private_key, recipient_id, message):
...
-                                    .format(recipient_id, message["samp.mtype"]))
-
-            self._launch_thread(target=self._notify_, args=(private_key,
-                                                            recipient_id,
-                                                            message))
+            if (
+                self._is_subscribed(
+                    self._public_id_to_private_key(recipient_id), message["samp.mtype"]
+                )
+                is False
```
Comment (astrofrog) [inline astropy/samp/hub.py:1053]: We should probably change this to ``not ...``

File: astropy/samp/hub.py [hunk 3]
```diff
@@ -1003,72 +1120,97 @@ def _notify_all_(self, sender_private_key, message):
...
         if private_key in self._private_keys:
-            if self._is_subscribed(self._public_id_to_private_key(recipient_id),
-                                   message["samp.mtype"]) is False:
-                raise SAMPProxyError(2, "Client {} not subscribed to MType {}"
-                                     .format(recipient_id, message["samp.mtype"]))
+            if (
+                self._is_subscribed(
+                    self._public_id_to_private_key(recipient_id), message["samp.mtype"]
+                )
+                is False
```
Comment (astrofrog) [inline astropy/samp/hub.py:1138]: Another instance that should be changed to ``not ...``

File: astropy/samp/hub.py [hunk 4]
```diff
@@ -1136,12 +1282,17 @@ def _reply_(self, responder_private_key, msg_id, response):
             return
 
         responder_public_id = self._private_keys[responder_private_key][0]
-        counter, hub_public_id, recipient_public_id, recipient_msg_tag = msg_id.split(";;", 3)
+        counter, hub_public_id, recipient_public_id, recipient_msg_tag = msg_id.split(
+            ";;", 3
+        )
```
Comment (astrofrog) [inline astropy/samp/hub.py:1287]: Cases like this are unfortunate, though not sure how we can write it more readably

File: astropy/samp/tests/test_web_profile.py [hunk 5]
```diff
@@ -68,10 +67,14 @@ def test_web_profile(self):
...
-        with get_readable_fileobj(f'http://localhost:{self.hub._web_port}/crossdomain.xml') as f:
+        with get_readable_fileobj(
+            f'http://localhost:{self.hub._web_port}/crossdomain.xml'
+        ) as f:
             assert f.read() == CROSS_DOMAIN
 
-        with get_readable_fileobj(f'http://localhost:{self.hub._web_port}/clientaccesspolicy.xml') as f:  # noqa: E501
+        with get_readable_fileobj(
+            f'http://localhost:{self.hub._web_port}/clientaccesspolicy.xml'
+        ) as f:  # noqa: E501
```
Comment (astrofrog) [inline astropy/samp/tests/test_web_profile.py:77]: Wonder if we can remove the noqa?

File: astropy/samp/tests/web_profile_test_helpers.py [hunk 6]
```diff
@@ -68,8 +70,9 @@ def _samp_hub(self):
         return self.proxy.samp.webhub
 
     def set_xmlrpc_callback(self, private_key, xmlrpc_addr):
-        raise NotImplementedError("set_xmlrpc_callback is not defined for the "
-                                  "web profile")
+        raise NotImplementedError(
+            "set_xmlrpc_callback is not defined for the " "web profile"
```
Comment (astrofrog) [inline astropy/samp/tests/web_profile_test_helpers.py:74]: Another string to fix

File: astropy/samp/tests/web_profile_test_helpers.py [hunk 7]
```diff
@@ -194,12 +197,14 @@ def register(self):
             result = self.hub.register("Astropy SAMP Web Client")
 
             if result["samp.self-id"] == "":
-                raise SAMPClientError("Registation failed - samp.self-id "
-                                      "was not set by the hub.")
+                raise SAMPClientError(
+                    "Registation failed - samp.self-id " "was not set by the hub."
```
Comment (astrofrog) [inline astropy/samp/tests/web_profile_test_helpers.py:201]: And another - quite a few in this file

**Review by astrofrog — APPROVED** [review]
As a former maintainer of astropy.samp, happy to approve this - a few small comments below in case anyone has time to fix, but not crucial

File: astropy/samp/__init__.py (hunk 1 above)
Comment (WilliamJamieson) [reply astropy/samp/__init__.py]: Fixed by the `--preview` flag!

File: astropy/samp/tests/test_web_profile.py (hunk 5 above)
Comment (WilliamJamieson) [reply astropy/samp/tests/test_web_profile.py:74]: This maybe stripped by pre-commit once we normalize the strings.

File: astropy/samp/tests/web_profile_test_helpers.py (hunk 6 above)
Comment (WilliamJamieson) [reply astropy/samp/tests/web_profile_test_helpers.py]: Fixed by the `--preview` flag!

File: astropy/samp/tests/web_profile_test_helpers.py (hunk 7 above)
Comment (WilliamJamieson) [reply astropy/samp/tests/web_profile_test_helpers.py]: Fixed by the `--preview` flag!

File: astropy/samp/tests/test_web_profile.py (hunk 5 above)
Comment (WilliamJamieson) [reply astropy/samp/tests/test_web_profile.py]: Looks like it was fixed!

File: astropy/samp/hub.py (hunk 3 above)
Comment (WilliamJamieson) [reply astropy/samp/hub.py]: Changed in commit 716d079. Feel free to revert it.

File: astropy/samp/hub.py (hunk 4 above)
Comment (WilliamJamieson) [reply astropy/samp/hub.py:1259]: Changed in commit 716d079. Feel free to revert it.

File: astropy/samp/hub.py (hunk 2 above)
Comment (WilliamJamieson) [reply astropy/samp/hub.py]: Changed in commit 716d079. Feel free to revert it.

**Review by nstarman — APPROVED** [review]
LGTM! Thanks @WilliamJamieson . I know it's a lot, but when this ``[.pre-commit-config.yaml](https://github.com/astropy/astropy/pull/13962/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9)`` is merged, rebasing the other PRs (or at least the ones I'm reviewing 😆).

**Comment by WilliamJamieson** [comment]
> LGTM! Thanks @WilliamJamieson . I know it's a lot, but when this `[.pre-commit-config.yaml](https://github.com/astropy/astropy/pull/13962/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9)` is merged, rebasing the other PRs (or at least the ones I'm reviewing 😆).

@nstarman I was planning on removing that suppression of string normalization checks, and normalizing the strings in this PR prior to merging it. I added it on a temporary basis so that I could continue to enforce the other pre-commit checks/fixes. If you would like I (or you) can push the same change temporarily to the PRs you are reviewing.

**Review by astrofrog — APPROVED** [review]
Looks good so far! Ready for string normalization.

**Comment by astrofrog** [comment]
@WilliamJamieson - this will need to be rebased due to a conflict

**Comment by WilliamJamieson** [comment]
> @WilliamJamieson - this will need to be rebased due to a conflict

Done

**Pull request merged**